src/GogeDao.sol

```diff
@@ -370,718 +476,955 @@ contract GogeDAO is Ownable {
+    /// @param _advocate address of wallet that we are removing their advocacy.
+    /// @param _pollNum the number of the poll the address is no longer an advocate for.
+    function _removeAdvocate(address _advocate, uint256 _pollNum) internal {
+        uint256 l = advocateFor[_advocate].length;
+        for (uint256 i = 0; i < l; i++) {
+            if (advocateFor[_advocate][i] == _pollNum) {
+                advocateFor[_advocate][i] = advocateFor[_advocate][--l];
+                advocateFor[_advocate].pop();
+            }
         }
     }
-    
-    // ---------- Polls ----------
 
-    /// @notice is used to create a new poll.
-    /// @param  _pollType enum type of poll being created.
-    /// @param  _change the matching metadata that will result in the execution of the poll.
-    function createPoll(PollType _pollType, Metadata memory _change) public {
-        _change.time1 = block.timestamp;
+    /// @notice An internal method for changing gateKeeping status.
+    /// @param _enabled status of gateKeeping.
+    function _setGateKeeping(bool _enabled) internal {
+        require(gateKeeping != _enabled, "Already set");
+        gateKeeping = _enabled;
+        emit GateKeepingModified(_enabled);
+    }
 
-        require(_change.time1 < _change.time2, "End time must be later than start time");
-        require(_change.time2.sub(_change.time1) >= minPeriod, "Polling period must be greater than 24 hours");
+    /// @notice An internal method for adding a team member address to the teamMembers array.
+    /// @param  _addr address of team member.
+    /// @param  _value is a team member.
+    function _setTeamMember(address _addr, bool _value) internal {
+        if(_value) {
+            (bool _isTeamMember, ) = isTeamMember(_addr);
+            if(!_isTeamMember) teamMembers.push(_addr);        
+        } else {
+            (bool _isTeamMember, uint8 s) = isTeamMember(_addr);
+            if(_isTeamMember){
+                teamMembers[s] = teamMembers[teamMembers.length - 1];
+                teamMembers.pop();
+            } 
+        }
+    }
 
-        emit ProposalCreated(pollNum, _pollType, _change.time1, _change.time2);
+    /// @notice An internal method for removing a poll from activePolls array.
+    /// @param _pollNum unique identifier for a poll.
+    function _removePollFromActivePolls(uint256 _pollNum) internal {
+        uint256 l = activePolls.length;
+        for (uint256 i = 0; i < l; i++) {
+            if (_pollNum == activePolls[i]) {
+                activePolls[i] = activePolls[--l];
+                activePolls.pop();
+            }
+        }
+    }
 
-        pollNum += 1;
-        pollTypes[pollNum] = _pollType;
-        pollMap[pollNum]   = _change;
+    /// @notice An internal method for updating a poll's end unix to current block.timestamp.
+    /// @param  _pollNum unique poll identifier.
+    function _updateEndTime(uint256 _pollNum) internal {
```

Comment:
Why is an extra method necessary for this purpose outside of manually ending polls?